src/App.jsx

```diff
@@ -354,28 +378,52 @@ function KanbanBoard() {
                             {provided.placeholder}
                             {user && (
                               <div className="add-task-button-container">
-                                <button
-                                  className="add-task-button"
-                                  onClick={() => {
-                                    const taskContent = prompt('Enter task content:');
-                                    if (taskContent?.trim()) {
-                                      const dueDate = new Date();
-                                      dueDate.setDate(dueDate.getDate() + 7); // Default to 1 week from now
-                                      addTask(column.id, {
-                                        id: crypto.randomUUID(),
-                                        content: taskContent,
-                                        createdAt: dueDate.toISOString(),
-                                        dueDate: dueDate.toISOString(),
-                                        labels: []
-                                      });
-                                    }
-                                  }}
-                                >
-                                  <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" width="16" height="16">
-                                    <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 4v16m8-8H4" />
-                                  </svg>
-                                  Add a card
-                                </button>
+                                {addingTaskToColumn === column.id ? (
+                                  <div className="inline-task-form">
+                                    <textarea
+                                      value={newTaskContent}
+                                      onChange={(e) => setNewTaskContent(e.target.value)}
+                                      placeholder="Enter a title for this card..."
+                                      autoFocus
+                                      onKeyDown={(e) => {
+                                        if (e.key === 'Enter' && !e.shiftKey) {
+                                          e.preventDefault();
+                                          handleAddTask(column.id);
+                                        } else if (e.key === 'Escape') {
+                                          setAddingTaskToColumn(null);
+                                          setNewTaskContent('');
+                                        }
+                                      }}
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const handleKeyDown = useCallback((columnId) => (e) => {
  if (e.key === 'Enter' && !e.shiftKey) {
    e.preventDefault();
    handleAddTask(columnId);
  } else if (e.key === 'Escape') {
    setAddingTaskToColumn(null);
    setNewTaskContent('');
  }
}, [handleAddTask]);

// Use in JSX:
onKeyDown={handleKeyDown(column.id)}
```
Memoize the keyboard event handler function to prevent unnecessary re-renders and improve performance



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;